packages/demo/webworker-lsp/webworker.ts

```diff
@@ -2,66 +2,79 @@ import {ServiceManager} from "ace-linters/build/service-manager";
     module: () => import("ace-linters/build/css-service"),
     className: "CssService",
     modes: "css"
 });
 manager.registerService("less", {
+    features: {completion: true, completionResolve: true, diagnostics: true, format: true, hover: true},
     module: () => import("ace-linters/build/css-service"),
     className: "CssService",
     modes: "less"
 });
 manager.registerService("scss", {
+    features: {completion: true, completionResolve: true, diagnostics: true, format: true, hover: true},
     module: () => import("ace-linters/build/css-service"),
     className: "CssService",
     modes: "scss"
 });
 manager.registerService("json", {
+    features: {completion: true, completionResolve: true, diagnostics: true, format: true, hover: true},
     module: () => import("ace-linters/build/json-service"),
     className: "JsonService",
     modes: "json",
 });
 manager.registerService("json5", {
+    features: {completion: true, completionResolve: true, diagnostics: true, format: true, hover: true},
     module: () => import("ace-linters/build/json-service"),
     className: "JsonService",
     modes: "json5",
 });
 manager.registerService("typescript", {
+    features: {completion: true, completionResolve: true, diagnostics: true, format: true, hover: true},
     module: () => import("ace-linters/build/typescript-service"),
     className: "TypescriptService",
     modes: "typescript|tsx|javascript|jsx",
 });
 manager.registerService("lua", {
+    features: {completion: false, completionResolve: false, diagnostics: true, format: false, hover: false},
     module: () => import("ace-linters/build/lua-service"),
     className: "LuaService",
     modes: "lua",
 });
 manager.registerService("yaml", {
+    features: {completion: true, completionResolve: true, diagnostics: true, format: true, hover: true},
     module: () => import("ace-linters/build/yaml-service"),
     className: "YamlService",
     modes: "yaml",
 });
 manager.registerService("xml", {
+    features: {completion: false, completionResolve: false, diagnostics: true, format: false, hover: false},
     module: () => import("ace-linters/build/xml-service"),
     className: "XmlService",
     modes: "xml",
 });
 manager.registerService("php", {
+    features: {completion: false, completionResolve: false, diagnostics: true, format: false, hover: false},
     module: () => import("ace-linters/build/php-service"),
     className: "PhpService",
     modes: "php"
 });
-/*manager.registerService("javascript", {
+manager.registerService("javascript", {
+    features: {completion: false, completionResolve: false, diagnostics: true, format: false, hover: false},
     module: () => import("ace-linters/build/javascript-service"),
     className: "JavascriptService",
     modes: "javascript",
-});*/
+});
 manager.registerService("python", {
+    features: {completion: false, completionResolve: false, diagnostics: true, format: false, hover: false},
```

Comment:
Why do we define features on service declaration instead of service itself. I did not see how the features object is used before loading  the service.

Reply:
We only initialize the Service instance upon the first request from the `LanguageProvider`, with the supported mode. At the time of creating these objects with features, we do not have any service instance available. This approach allows users to set up the system more easily, as they can choose any combination of supported features from different services simply by specifying them in the web worker. Features should not be equated with service capabilities. Another method to toggle features during runtime is to use `toggleServiceFeatures`; however, this goes through the `messageController` to the web worker's `ServiceManager`, making it, in my opinion, a less preferable way to toggle functionality.